x/masterchef/keeper/abci.go

```diff
@@ -236,7 +236,9 @@ func (k Keeper) UpdateLPRewards(ctx sdk.Context) error {
 		}
 
 		// Distribute Eden
-		k.UpdateAccPerShare(ctx, pool.PoolId, ptypes.Eden, newEdenAllocatedForPool.TruncateInt())
+		if pool.EnableEdenRewards {
```

Comment:
If we are considering eden rewards for a set of pools, then `newEdenAllocatedForPool` this value needs to be updated too.
In this file we are calculating proxyTVL and proxyTVLTotal, to calculate ratio.
We need to consider this while calculating newEdenAllocatedForPool (exclude or include pool)

Reply:
1. [Here](https://github.com/elys-network/elys/blob/3a77b329127f72d76f6b6c02fd7efa855ca72768/x/masterchef/keeper/abci.go#L174) we are calculating proxyTVL considering all pools, so we need to update this value to set of pools we are allowing rewards for
2. The reason is [here](https://github.com/elys-network/elys/blob/3a77b329127f72d76f6b6c02fd7efa855ca72768/x/masterchef/keeper/abci.go#L206), we calculate ratio, so if we want to skip some pools, they should not reflect in proxyTVL